### PR TITLE
Array or List in query string does not get parsed #7712

### PR DESCRIPTION
**Pull request by kishanAnem**
After resolving conflicts.
[35955df](https://github.com/aspnet/Mvc/commit/35955df9d84afbb874e0278f3ab58da88da868d1)

### CODE REVIEW

**Comment by kishanAnem** [comment]
Hi @dougbu 
 I think now you can see my changes.

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ApiBehaviorApplicationModelProviderTest.cs [hunk 1]
```diff
@@ -620,6 +621,75 @@ public void PreservesBindingSourceInference_ForFromQueryParameterOnComplexType_W
             Assert.Equal("gps", bindingInfo.BinderModelName);
         }
 
+        [Fact]
+        public void PreservesBindingSourceInference_ForFromQueryParameterOnCollectionType()
```
[the remark this answers is not in the document]
Comment (kishanAnem) [reply test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ApiBehaviorApplicationModelProviderTest.cs]: @dougbu Will add send this again, thanks for help😊

**Comment by kishanAnem** [comment]
Hi @dougbu ,

I added a test case.

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ApiBehaviorApplicationModelProviderTest.cs [hunk 2]
```diff
@@ -482,6 +483,31 @@ public void OnProvidersExecuting_PreservesBindingInfo_WhenInferringFor_Parameter
             Assert.Equal("top", bindingInfo.BinderModelName);
         }
 
+        [Fact]
+        public void OnProvidersExecuting_PreservesBindingInfo_WhenInferringFor_ParameterWithModelBinder_WithoutExplicitName_OnCollection()
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ApiBehaviorApplicationModelProviderTest.cs]: This is not the addition I was looking for in my earlier [comment](https://github.com/aspnet/Mvc/pull/7967/files#r197895856). Need to exercise your changes to `InferBoundPropertyModelPrefixes(...)`. That is, model the new test after `InferBoundPropertyModelPrefixes_SetsModelPrefix_ForComplexTypeFromValueProvider()` but use a property of a collection type (array, list, whatever).

**Comment by kishanAnem** [comment]
welcome @dougbu, i did 50% work. remaining with your help. thank you very much @dougbu.

**Comment by VictorioBerra** [comment]
I think that this broke my code when I updated to SDK 2.1.403 and I have been pulling my hair out trying to determine why.

Here is my controller:

```
[ApiController]
public class ProductsController : ControllerBase
{
    public async Task<List<ProductViewModel>> PutMany([FromBody]List<ProductUpdateViewModel> products)
    => await _genericCRUDService.UpdateRangeAsync(products);
}
```

I think when you guys did this: https://github.com/aspnet/Mvc/commit/5e20c31#diff-c693720b853e696ccec4442fb4f072a3R259

All my controllers that accept a `List<ComplexType>` now get treated as if they are NOT complex types.

I found that applying `[FromBody]` fixes my code. I thought that using `[ApiController]` attribute was supposed to prevent me from needing to do that kind of think everywhere.

**Comment by pranavkm** [comment]
@VictorioBerra we're tracking the regression via https://github.com/aspnet/Mvc/issues/8536

**Pull request merged**